src/libYARP_dev/src/ImplementControlMode.cpp

```diff
@@ -90,26 +99,29 @@ bool ImplementControlMode::setControlMode(const int j, const int mode)
 
 bool ImplementControlMode::setControlModes(const int n_joint, const int *joints, int *modes)
 {
-    int *tmp_joints=new int [nj];
+    if(!castToMapper(helper)->checkAxesIds(n_joint, joints))
+        return false;
+
+    Buffer<int> buffValues  = buffManager->getBuffer();
+
     for(int idx=0; idx<n_joint; idx++)
     {
-        MJOINTIDCHECK_DEL
-        tmp_joints[idx] = castToMapper(helper)->toHw(joints[idx]);
+        buffValues[idx] = castToMapper(helper)->toHw(joints[idx]);
     }
-    bool ret = raw->setControlModesRaw(n_joint, tmp_joints, modes);
+    bool ret = raw->setControlModesRaw(n_joint, buffValues.getData(), modes);
 
-    delete [] tmp_joints;
+    buffManager->releaseBuffer(buffValues);
     return ret;
 }
 
 bool ImplementControlMode::setControlModes(int *modes)
 {
-    int *modes_tmp=new int [nj];
-    for(int idx=0; idx<nj; idx++)
+    Buffer<int> buffValues  = buffManager->getBuffer();
+    for(int idx=0; idx<castToMapper(helper)->axes(); idx++)
```

Comment:
I would simplify the code using nj instead castToMapper(helper)->axes().
Same in all other interfaces.
Additionally, we are in the process of replacing all these int to size_t.

Reply:
just readable.. I can't find any good reason to ask to some else an info that I already know...